eclipse/src/saros/editor/annotations/SarosAnnotation.java

```diff
@@ -73,31 +78,46 @@ public static Color getUserColor(int colorID) {
   }
 
   /**
-   * Returns the color that corresponds to a user.
+   * Returns the color for the given annotation type and user color id.
    *
    * <p><b>Important notice:</b> Every returned color instance allocates OS resources that need to
    * be disposed with {@link Color#dispose()}!
    *
    * @param type
-   * @param colorID
+   * @param colorId
    * @return the corresponding color or a default one if no color is stored
    */
-  protected static Color getColor(String type, int colorID) {
-
-    type = getNumberedType(type, colorID);
+  public static Color getColor(final String type, final int colorId) {
 
-    AnnotationPreferenceLookup lookup = EditorsUI.getAnnotationPreferenceLookup();
+    final String typeToLookup = getNumberedType(type, colorId);
 
-    AnnotationPreference ap = lookup.getAnnotationPreference(type);
+    final AnnotationPreference ap =
+        EditorsUI.getAnnotationPreferenceLookup().getAnnotationPreference(typeToLookup);
 
     if (ap == null)
-      throw new RuntimeException(
-          "could not read color value of annotation '" + type + "' because it does not exists");
+      throw new IllegalArgumentException(
+          "could not read color value of annotation '"
+              + typeToLookup
+              + "' because it does not exists");
+
+    RGB rgb =
+        PreferenceConverter.getColor(EditorsUI.getPreferenceStore(), ap.getColorPreferenceKey());
 
-    return new Color(Display.getDefault(), ap.getColorPreferenceValue());
+    if (rgb == PreferenceConverter.COLOR_DEFAULT_DEFAULT) { // NOPMD
+      rgb = ap.getColorPreferenceValue();
+    }
+
+    if (rgb == null)
+      throw new IllegalArgumentException(
+          "annotation '" + typeToLookup + "' does not have a default color");
```

Comment:
How about stating that we encountered an unknown annotation color type (under the assumption that every valid type has a default value)?

Reply:
if you call this with ("foo.bar", 4711) it will be translated to foo.bar.Default

I could also throw "you fucked up the plugin.xml". I think this is more helpful.

It is just a look up of

      <specification
            annotationType="saros.annotations.viewport.default"
            colorPreferenceKey="saros.annotations.viewport.color.default"
            colorPreferenceValue="128,128,128"
            contributesToHeader="true"
            highlightPreferenceKey="saros.annotations.viewport.highlight.default"
            highlightPreferenceValue="false"
            includeOnPreferencePage="true"
            label="DPP Default Viewport"
            overviewRulerPreferenceKey="saros.annotations.viewport.overview.default"
            overviewRulerPreferenceValue="true"
            presentationLayer="1"
            textPreferenceKey="saros.annotations.viewport.text.default"
            textPreferenceValue="false"
            textStylePreferenceValue="NONE"
            verticalRulerPreferenceKey="saros.annotations.viewport.ruler.default"
            verticalRulerPreferenceValue="false"/>
      <specification

if you do add a ColorPreferenceValue this exception will be thrown